### ISSUE TITLE
How to authenticate user in API functions called by getInitialProps from the backend

### ISSUE THREAD

**Issue by fromi** [auth0/nextjs-auth0#46]
### Describe the problem you'd like to have solved

We can call an API function from a page getInitialProps function (example from Next.js tutorial using isomorphic-unfetch: https://nextjs.org/learn/basics/fetching-data-for-pages/fetching-batman-shows)
In this case, auth0.getSession does not work, because the authentication cookie is not passed around the backend functions with isomorpic-unfetch.

### Describe the ideal solution

Some helper function to pass around the session cookie between API endpoints?
At least some documentation about the problem in the README section and a recommended way to handle it.

## Alternatives and current work-arounds

I finally managed to forward the session cookie thanks to this post: https://github.com/apollographql/apollo-client/issues/5089#issuecomment-540605542

Here is my version:
```
  const cookieForwardingFetch = (url: RequestInfo, init: RequestInit) => {
    const headers = req.headers.cookie ? {...init.headers, 'Cookie': req.headers.cookie} : init.headers
    return fetch(url, {...init, headers: headers})
  }
```

Thanks for the great work!

**Comment by tsaiDavid** [comment]
I'd like to understand the best method in working with an Apollo Client in Next.JS as well in conjunction with this SDK

**Comment by samuelhnrq** [comment]
Here am I, the issue necromancer. 

I was actually looking for something completely unrelated, but I never even thought of that, and the more I think about it the sillier I feel, (not using getInitialProps kinda beats the purpose of using SSR 
 amiright 😆) but, I don't know when this issue was posted but nowadays isn't this achieved by using the [context object](https://nextjs.org/docs/api-reference/data-fetching/getInitialProps#context-object)'s req and res on `auth0.tokenCache`?

**Comment by Widcket** [comment]
Hi everyone, we've just released v1.0.0-beta.0 that includes support for protecting server-side rendered pages with `getServerSideProps`. Please take a look at [the example](https://github.com/auth0/nextjs-auth0/blob/beta/EXAMPLES.md#protecting-a-server-side-rendered-ssr-page). Note that the beta supports only Next 10+, though.
